# analyses of intrahost and interhost sapovirus evolution
Studies of sapovirus intra-genotypic evolution patterns, within and between-host evolution may contribute to a better understanding of worldwide sapovirus epidemiology and evolution. 
I mined whole genbank sapovirus capsid data and uncovered different patterns of evolution between different genotypes. Sapovirus genome was then sequenced using a multiplex amplicon sequencing protocol published by Quick et al, Nat. protocols (https://www.nature.com/articles/nprot.2017.066). I compared intrahost and inter-host mutations / single nucleotide variations (SNVs) samples from Peru and Miyagi-Japan between GI.1 and GI.2 viral populations using a method proposed by Grubaugh et al, (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1618-7). 

*Required softwares for NGS analyses*
- samtools 1.10 (Using htslib 1.10.2)
- iVar version 1.3.1

*NB*: All analyses were performed on Ubuntu 18.04.5LTS operating system, memory 64GB, Intel Xeon(R) CPU v3 @ 1.6GHZ x 12, Graphics GeForce RTX 2080/PCIe/SSE2. 
