# sapovirus-evolution
Studies of sapovirus intra-genotypic evolution patterns, within and between-host evolution may contribute to a better understanding of worldwide sapovirus epidemiology and evolution. 
I mined whole genbank sapovirus capsid data and uncovered different patterns of evolution between different genotypes. Sapovirus genome was then sequenced using a multiplex amplicon sequencing protocol published by Quick et al, Nat. protocols (https://www.nature.com/articles/nprot.2017.066). That protocol of Quick et al., was later proven useful for virus single nucleotide polymorphism detection in 2019 (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1618-7), because of that I decided to use it my study because i found that it was the most sensitive method for amplifying sapovirus genomes in fecal shedding samples (Unpublished data). 
I compared time-wise single nucleotide variations of GI.1 and GI.2 populations within infected hosts.
I also compared intrahost genetic diversity of GI.1 and GI.2 populations. 
I made an attempt to classify sapovirus into larger groups called " immunotypes" and performed additional statistical analyses. 
